test/core/test/mocks/mocked-no-mocks.test.ts

```diff
@@ -0,0 +1,22 @@
+import { expect, test, vi } from 'vitest'
+import { mockedA } from '../../src/mockedA'
+import { mockedB } from '../../src/mockedB'
+
+vi.mock('../../src/mockedA.ts')
+vi.mock('../../src/mockedB.ts')
+
+test('testing mocking module without __mocks__', () => {
+  mockedA()
+  expect(mockedA).toHaveBeenCalledTimes(1)
+})
+
+test('mocking several modules work', () => {
+  // Cannot unmock a module that has been mocked via a file path
+  // See more: https://stackoverflow.com/a/56512217
+  // vi.mocked(mockedB).mockRestore()
```

Comment:
it doesnt unmock it, it just restores implementation to default one

if jest can't do it, why we shouldnt? I think previous implementation was right. our spyOn can be called anywhere

Reply:
and the idea of this test was to check if `mockedA` was called by `mockedB`, now it's lost since by default `mockedB` does nothing